src/IntegrationTests/IncludeMembers.cs

```diff
@@ -1630,4 +1630,105 @@ public void Should_flatten()
             result.ShouldBeOfType<Destination>();
         }
     }
+}
+public class IncludeOnlySelectedMembers : IntegrationTest<IncludeOnlySelectedMembers.DatabaseInitializer>
```

Comment:
This needs to use an expression with `FirstOrDefault` to use the let clause.

Reply:
I changed the test like you requested locally but it is broken, although not a regression. the FirstOrDefault let expression brings in the whole object instead of the requested properties. 

I am investigating if this can be fixed by adding a projection that only includes the required properties

Reply:
found a solution

Reply:
But that just brings back the one piece of code you removed.

Reply:
the original code was not preserving type information. 
if you have suggestions on what code should be restored please let me know. then i will try to use that code instead.

Reply:
also i tried to repro on master but strangely there the query is correct even if the let clause contains the whole object included by firstOrDefault. im still investigating whats the difference there...

Reply:
I was able to repro the issue.
as soon as the let statement contains a conditional statement entity framework selects all the properties instead of the required properties.

https://github.com/JanEggers/AutoMapper/tree/queryContainsMoreThanRequiredProperties